packages/pkp-walletconnect/src/lib/pkp-walletconnect.ts

```diff
@@ -0,0 +1,638 @@
+import {
+  SupportedETHSigningMethods,
+  ethRequestHandler,
+  isEthRequest,
+} from '@lit-protocol/pkp-ethers';
+import { LIT_CHAINS } from '@lit-protocol/constants';
+import { Core } from '@walletconnect/core';
+import {
+  IWeb3Wallet,
+  Web3Wallet,
+  Web3WalletTypes,
+} from '@walletconnect/web3wallet';
+import {
+  formatAccountWithChain,
+  getSdkError,
+  parseChainId,
+} from '@walletconnect/utils';
+import {
+  ErrorResponse,
+  formatJsonRpcError,
+  formatJsonRpcResult,
+  JsonRpcResponse,
+} from '@json-rpc-tools/utils';
+import {
+  CoreTypes,
+  ISignClient,
+  SessionTypes,
+  SignClientTypes,
+} from '@walletconnect/types';
+import { PKPClient } from '@lit-protocol/pkp-client';
+import { PKPBase } from '@lit-protocol/pkp-base';
+
+const DEFAULT_RELAY_URL = 'wss://relay.walletconnect.com';
+
+export interface InitWalletConnectParams
+  extends Omit<Web3WalletTypes.Options, 'core'> {
+  projectId: string;
+  relayUrl?: string;
+}
+
+export class PKPWalletConnect {
+  // WalletConnect client
+  private client: IWeb3Wallet | undefined;
+  // List of PKPClients
+  private pkpClients: PKPClient[] = [];
+  // Supported chains
+  private supportedChains: string[] = ['eip155'];
+
+  // For logging
+  private readonly debug: boolean = false;
+  private readonly PREFIX = '[PKPWalletConnect]';
+  private readonly orange = '\x1b[33m';
+  private readonly reset = '\x1b[0m';
+  private readonly red = '\x1b[31m';
+
+  constructor(debug?: boolean) {
+    this.debug = debug || false;
+  }
+
+  /**
+   * Initializes the WalletConnect client
+   *
+   * @param {InitWalletConnectParams} params
+   * @param {string} params.projectId - The WalletConnect project ID
+   * @param {string} [params.relayUrl] - The WalletConnect relay URL
+   */
+  public async initWalletConnect(
+    params: InitWalletConnectParams
+  ): Promise<void> {
+    let coreOpts: CoreTypes.Options = {
+      projectId: params.projectId,
+      relayUrl: params.relayUrl || DEFAULT_RELAY_URL,
+    };
+    if (this.debug) {
+      coreOpts = { ...coreOpts, logger: 'debug' };
+    }
+    const core = new Core(coreOpts);
+    this.client = await Web3Wallet.init({
+      core,
+      metadata: params.metadata,
+      name: params.name,
+    } as Web3WalletTypes.Options);
+  }
+
+  /**
+   * Pair with the given URI received from a dapp
+   */
+  public pair: IWeb3Wallet['pair'] = async (params) => {
+    this.client = this._isWalletConnectInitialized(this.client);
+    return await this.client.pair(params);
+  };
+
+  /**
+   * Parse the session proposal received from a dapp, construct the session namespace,
```

Comment:
what's a session namespace?

Reply:
namespaces are CAIP-standardized objects that describe what accounts, chains, methods are requested and used during a dapp-wallet session facilitated by WalletConnect